### ISSUE TITLE
Songs: Song language is not copied to lyrics language + can be overwritten by lyrics language detection

### ISSUE THREAD

**Issue by szc126** [VocaDB/vocadb#1715]
### Website (required)
VocaDB

### Describe the bug (required)
1. Song language is not copied to the lyrics language field.
2. After adding lyrics, the detected language will overwrite the song language.

Example: https://vocadb.net/Song/Versions/414349

### To Reproduce (optional)
1. Open a song edit page
2. Set the song language
3. Add lyrics of a different language
4. Wait for lyrics language to be detected (automatically change from "Other/Unknown" (!!))
5. Save the song entry

### Expected behavior (optional)
1. Lyrics language is autofilled with the song language.
2. Song language is obeyed and not overwritten

### Additional context (optional)
Related to #1695?

### Checklist (required)

Fill out the checklist. Use ~~strikethroughs~~ or leave fields empty if you don't know or are unable to check/test.

- [x] Searched **[VocaDB/vocadb](https://github.com/VocaDB/vocadb/issues)** for duplicate issues.
- [x] Tested on the **[beta site](https://beta.vocadb.net)**.
- [ ] Tested on different browsers (e.g. **Firefox**, **Chrome**, **Safari**, **Edge**).
- [ ] Checked in **private mode**.
- [ ] Cleared **browser cache** and **cookies**.
- [ ] Checked the **DevTools console**.
- [ ] Checked the **syslog** (mods+ only).

**Comment by CatgirlFrostmoon** [comment]
Not a bug, exactly: this, once again, helps us handle the oddities with the Chinese language cluster. Mandarin is a spoken language, Chinese (Traditional or Simplified) is a written script, and having these separate stops things from either just being labelled as "Chinese" (t'would amalgamate Mandarin, Cantonese, Hokien, etc. into one language, which isn't correct) or from things like Mandarin ending up as the lyrical language.

**Comment by szc126** [comment]
I disagree with this being helpful for the Chinese situation. As a database handling vocal songs, including very few dialect songs (essentially a curiosity, 講得唔好聽), determining which "Chinese" a song is in has essentially 0% friction.

"Chinese" as a song language should really be abolished and replaced uniformly with Mandarin. I reckon that the number of dialect songs misclassified is less than 10, usually being conspicuous; we can probably run a search on video titles, and the only possibly thorny case I am actively aware of is a song that has a Sinitic conlang, <https://vocadb.net/S/743612>, but I've already classified it as "Fictional".

I have no objection to Mandarin (or Cantonese, etc.) being the lyrics language for songs, either.

It would only be a mild inconvenience when "changed language" blocks lyrics inheritance for songs where someone transfers the lyrics character-by-character, and we could create a tag instead for those cases (<https://vocadb.net/S/257058>, <https://vocadb.net/S/596606>), the opposite of "changed lyrics", to force lyrics inheritance. From my impression, most Mandarin→dialect "changed language" songs have at least minor vocabulary changes.

And if we want to worry about Cantopop and 書面語 (lexically and grammatically Standard Chinese = Mandarin lyrics) and 口語 (lexically and grammatically spoken Cantonese; do we have any?), we can create tags for those instead, like <https://vocadb.net/T/9522/literary-japanese> (lexically and gramatically Old / pre-modern Japanese) and <https://vocadb.net/T/9830/modern-japanese>.

We also have the issue where societal and technological pressures compel people to transcribe lyrics in Standard Chinese-ized (Mandarin-ized) form, although I'm not sure if any cases are on VocaDB specifically, since the people who do bother with forcing voice synthesizers to cross languages tend to be nerdy.

**Comment by CatgirlFrostmoon** [comment]
"We also have the issue where societal and technological pressures compel people to transcribe lyrics in Standard Chinese-ized (Mandarin-ized) form" <-- This was basically what I was trying to reference, though it's interesting to hear that we might not really be dealing with it. I'm aware that separate Chinese languages can basically read the same script, but I also know that Cantonese has its own writing system as well.

If swapping Chinese to Mandarin will work, then I don't actually have any objections; I mostly just want to make sure it's all accurate, and that we're not conglomerating a group of languages together as if they're one. That, and I want to avoid the goofiness seen on stuff like Crunchyroll where spoken languages are listed as "Simplified Chinese". XD

**Comment by szc126** [comment]
Forgot to write: arguably "Chinese" could be useful for song \*translations\*, being entirely written, but it's also still guaranteed to be in the standard Chinese language understood throughout China (= Mandarin) so it's probably also not a concern; marking these as Mandarin would keep the language list simple.

Maybe someone having too much fun (<https://lyricstranslate.com/en/ジャンク-垃圾.html>) would write a Literary Chinese translation? but the amount of creative liberty that would be taken would probably make it out-of-scope for VocaDB, similar to translyrics. (Unrelated: maybe it would be nice to have a 文言文 Literary Chinese tag for song lyrics? or maybe poetic language is too normalized for Chinese music for this to be useful)

**Comment by szc126** [comment]
https://vocadb.net/Song/Versions/414349; I would like to request a database scan of edits matching `Updated properties (Lyrics, Languages)` if possible.

**Comment by CatgirlFrostmoon** [comment]
I believe adding lyrics brings up that same edit list; in other words, we're going to get a lot of results attached to this one request, and not all will be relevant. That, and I'm really quite uncomfortable with the idea of setting a perfect loop up, since we'd then have zero way of breaking it in the case of written cyphers or similar.

EDIT: Yep, cyphered lyrics were just mentioned on the Discord. That absolutely makes this inviable without a way of killing the infinite loop.

**Comment by szc126** [comment]
Well superficially the message would be `Updated properties (Lyrics, Languages)`, but preferably we could then filter those specifically for cases where the language has changed from one value to another (instead of from null to any value).

I'm also not sure what you mean by "perfect loop". It would simply improve usability and be less of a nasty surprise if
1. the "Basic info" language field is synchronized with the "Lyrics" language field (a manual change to one is always reflected in the other direction; currently it is unidirectional);
2. the system does not overwrite either field without asking for confirmation.

As for cyphers, if the "Lyrics" language is stored as, say, "en-Cypher" ("English - Cyphered" or some other [actual standard-conformant script code](https://en.wikipedia.org/wiki/ISO_15924)), instead of simply "Cypher", we do not have inconsistency. Same for storing Chinese (Mandarin) lyrics as "cmn" instead of "zh".

**Comment by szc126** [comment]
> transcribe lyrics in Standard Chinese-ized (Mandarin-ized) form

https://vocadb.net/S/953037; one line in Zhuang but simply presented in Chinese translation as `这边唱那边和`.

**Comment by CatgirlFrostmoon** [comment]
The perfect loop is that if this goes into affect, we have *no* way of not having Language and Lyrics Language be identical. This would inevitably lead to the same issues as are affecting us with inheritance, where not having an "out" will cause niche cases to become pretty noticeable problem zones.

**Comment by szc126** [comment]
Are there still cases we haven't discussed where such a situation would be necessary? Sorry if I missed anything.
